### DISCUSSION TITLE
Include cisco.nd collection into Ansible package

### DISCUSSION THREAD

**Discussion by lhercot** [ansible-collections/ansible-inclusion#77] [First review in progress]
The Cisco ND collection provides modules to manage and automate the configuration of Cisco Nexus Dashboard. It also provides the ND HTTPAPI plugin used by other collections to communicate with Nexus Dashboard.

Ansible Galaxy: https://galaxy.ansible.com/cisco/nd
GitHub: https://github.com/CiscoDevNet/ansible-nd
Issues Tracker: https://github.com/CiscoDevNet/ansible-nd/issues
GitHub Handles: @lhercot @akinross
Is the collection part of Automation Hub: Yes
We meet [Collection Requirements](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html): Yes

**Comment by Andersson007** [comment]
@lhercot hello, thanks for submitting the request. I changed the category to New collection reviews as it's a correct one for new requests. UPDATE: now it's "First review in progress" respectively

Here's my review

# Ansible Collections Checklist (short version)

_For details about the following points, refer to the [Collection Requirements](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html)._

Every comment should say whether the reviewer expects it to be addressed, or whether it's optional.

Note for reviewers: If you don't know how to check any of the points below, please ask maintainers of the collection you're reviewing or a [Steering Committee member](https://docs.ansible.com/ansible/devel/community/steering/community_steering_committee.html#current-steering-committee-members) for clarifications in comments of corresponding inclusion discussion.

**Public availability and communication:**
- [x] published on [Ansible Galaxy](https://galaxy.ansible.com) with version 1.0.0 or later
- [x] have a Code of Conduct (CoC) compatible with the [Ansible Code of Conduct (CoC)](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html)
- [x] have a publicly available issue tracker that does not require a paid level of service to create an account or view issues
- [x] have a public git repository
- [x] releases of the collection are tagged in its repository

**Standards and documentation:**
- [x] adheres to [semantic versioning](https://semver.org/)
- [x] follows [licensing rules](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#collection-licensing-requirements)
- [ ] follows the [Ansible documentation standards](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html) and the [style guide](https://docs.ansible.com/ansible/devel/dev_guide/style_guide/index.html#style-guide)
  **MUST FIX:** check mode support is not reflected in docs  
- [ ] follows [development conventions](https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_best_practices.html); as well as these other requirements:
  - [x] modules satisfy the concept of [idempotency](https://docs.ansible.com/ansible/latest/reference_appendices/glossary.html#term-Idempotency>)
  - [ ] modules that only gather information are named `<something>_info`
  **MUST FIX:** there are modules that do it but ther names don't satisfy this requirement
  - [x] modules that return `ansible_facts` are named `<something>_facts` and do not return non-facts
  - [ ] other modules must not allow querying information using specific `state` option values, or similar mechanisms (like `state=get` or `state=query`).  These features should be moved to `<something>_info` or `<something>_fact` modules.
  **MUST FIX:** modules use "state: query" to query information. This functionality should be moved out to _info module and the "query" choice removed from the module.  
  - [x] `check_mode` is supported in all `_info` and `_facts` modules
- [ ] supports [all Python versions supported by all ansible-core versions its supports](https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix). If it does not, read the [full guidelines](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#python-compatibility) to see if you qualify for an exception and document the unsupported [Python versions](https://docs.ansible.com/ansible/devel/reference_appendices/release_and_maintenance.html#ansible-core-support-matrix) in the collection ``README.md`` and in every module and plugin (or in doc fragments)
  **COMMENT:** does the collection support all Python versions required ^? Anyway, i would suggest including this info to README explicitly
- [x] only uses the [allowed plugin types](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#modules-plugins) in the `plugins/` directory
- [x] has `README.md`
- [x] FQCNs are used for all plugins and modules including `ansible.builtin.` for builtin ones from ansible-core in all their appearances in the documentation, examples, and return sections, and in `extends_documentation_fragment:`s.
- [x] documentation and return sections use `version_added:` containing the *collection* version for which an option, module or plugin was added (except cases when they were added in the very first release of the collection)
- [x] public plugins, roles and playbooks do not use files outside of `meta/`, `plugins/`, `roles/`, and `playbooks/`

**Collection management:**
- [x] `galaxy.yml` having `tags` field set
- [ ] collection dependencies must have a lower bound on the version which is at least 1.0.0, and are all part of the `ansible` package
  **MUST FIX:** Please set the lower bound for ansible.netcommon explicitly 
- [ ] `meta/runtime.yml` defines the minimal version of Ansible which the collection works with
  **SHOULD FIX:** I see requires_ansible: '>=2.15.0' but in README it's 2.14+. Please update README if it's irrelevant (and in other places if needed).
- [x] has changelog, preferably with `changelogs/changelog.yaml`
  ~~SHOULD FIX: CHANGELOG.rst and changelogs/changelog.yaml don't match~~
- [x] collection repository should not contain any large objects (binaries) comparatively to the current Galaxy tarball size limit of 20 MB. For example, package installers for testing purposes shouldn't be added.
- [x] collection repository should not contain any unnecessary files like, for example, temporary files. Temporary files should be added to `.gitignore`.

**Tests:**

Note for reviewers: If you don't know how to check the points below, please ask maintainers of the collection you're reviewing how you can do it.
* In most cases, CI is set up via GitHub Actions.
* Check `.yml` files in the `.github/workflows` directory. There must be at least the `sanity` section under `jobs` containing the `ansible-test sanity` command running against all supported ansible-core versions that must be also listed there, for example, `- stable-2.15, - stable-2.16, - stable-2.17`.
* Check workflow runs by clicking the `Actions` tab in the repository's page - you're interested in `Scheduled` runs, runs against release commits and runs against pull requests.
* If there are no workflows in the `Actions` tab, ask the collection maintainers how CI is implemented.

- [x] passed `ansible-test sanity`
- [x] if `test/sanity/ignore*.txt` exists, it MUST not contain error codes listed in the [list of errors that must not be ignored](https://docs.ansible.com/ansible/devel/community/collection_contributors/collection_requirements.html#ci-testing)
- [x] has CI tests up and running against each of the "major versions" of `ansible-base`/`ansible-core` that the collection supports
- [x] all CI tests MUST run against every pull request
- [x] all CI tests MUST run regularly (nightly, or at least once per week)
- [x] sanity tests MUST run against a commit that releases the collection; if they don't pass, the collection won't be released

**Comment by lhercot** [comment]
@Andersson007 Thanks for the review.

I am not sure to understand the following:
```
 has changelog, preferably with changelogs/changelog.yaml
SOULD FIX: CHANGELOG.rst and changelogs/changelog.yaml don't match
```

The CHANGELOG.rst is generated by using antsibull we do run yamlfix after antsibull run on changelog.yaml to comply with sanity requirements of YAML format as antsibull uses a different YAML format which at some point was breaking our sanity.
Is that the issue or am I missing something else?

**Reply by Andersson007** [reply]
@lhercot now i see they look like they match, sorry for confusion, i've marked this point as OK

**Reply by lhercot** [reply]
Great! Thanks!
